config.js

```diff
@@ -272,144 +265,75 @@ exports.updateFromArgv = function() {
         return false;
     };
     if (argv.ogspv) {
-        const ogsPv = argv.ogspv.toUpperCase();  // being case sensitive tolerant
-        checkUnsupportedOgspvAI(ogsPv, ogsPvAIs);
-        exports.ogspv = ogsPv;
+        exports.ogspv = argv.ogspv.toUpperCase();
     }
 
-    /* 2) specifc r_u cases :*/
-    if (argv.minrank && !argv.minrankranked && !argv.minrankunranked) {
-        exports.minrank = parseRank(argv.minrank);
-    }
-    if (argv.minrankranked) {
-        exports.minrankranked = parseRank(argv.minrankranked);
-    }
-    if (argv.minrankunranked) {
-        exports.minrankunranked = parseRank(argv.minrankunranked);
-    }
-    if (argv.maxrank && !argv.maxrankranked && !argv.maxrankunranked) {
-        exports.maxrank = parseRank(argv.maxrank);
-    }
-    if (argv.maxrankranked) {
-        exports.maxrankranked = parseRank(argv.maxrankranked);
-    }
-    if (argv.maxrankunranked) {
-        exports.maxrankunranked = parseRank(argv.maxrankunranked);
-    }
-
-    if (argv.bans) {
-        for (const user of argv.bans.split(',')) {
-            exports.banned_users[user] = true;
-        }
-    }
-    if (argv.bansranked) {
-        for (const user of argv.bansranked.split(',')) {
-            exports.banned_users_ranked[user] = true;
-        }
-    }
-    if (argv.bansunranked) {
-        for (const user of argv.bansunranked.split(',')) {
-            exports.banned_users_unranked[user] = true;
-        }
-    }
-
-    if (argv.boardsizes) {
-        for (const boardsize of argv.boardsizes.split(',')) {
-            if (boardsize === "all") {
-                exports.allow_all_boardsizes = true;
-            } else {
-                exports.allowed_boardsizes[boardsize] = true;
-            }
-        }
-    }
-    if (argv.boardsizesranked) {
-        for (const boardsizeranked of argv.boardsizesranked.split(',')) {
-            if (boardsizeranked === "all") {
-                exports.allow_all_boardsizes_ranked = true;
-            } else {
-                exports.allowed_boardsizes_ranked[boardsizeranked] = true;
-            }
-        }
-    }
-    if (argv.boardsizesunranked) {
-        for (const boardsizeunranked of argv.boardsizesunranked.split(',')) {
-            if (boardsizeunranked === "all") {
-                exports.allow_all_boardsizes_unranked = true;
-            } else {
-                exports.allowed_boardsizes_unranked[boardsizeunranked] = true;
+    // r_u exports
+    /* 2) specific r_u cases:*/
+    for (const familyNameString of ["minrank", "maxrank"]) {
+        for (const argNameString of getArgNameStringsGRU(familyNameString)) {
```

Comment:
we can make it into a function, it removes the nested loop and is simpler, yes

but for the `this.argv`, i dont like it because we dont use the argv except in config, also, we dont pass argv that much so maybe its fine to keep passing argv instead of adding extra things to check about

Reply:
but then we'd want to do it also with allowed families and other arg exports no ?

Reply:
> but for the this.argv, i dont like it because we dont use the argv except in config, also, we dont pass argv that much so maybe its fine to keep passing argv instead of adding extra things to check about

yes that's why i said a local field for config.js, not one we export.

Reply:
> but then we'd want to do it also with allowed families and other arg exports no ?

you mean making it a function we call multiple times? Yes, see my other comments below.

Reply:
i'll try to do it for all functions then :)

Reply:
i fixed it like this:
https://github.com/online-go/gtp2ogs/pull/230/commits/0fe038af02b2297d23b070ef580319d585a47733
in pr #221 it will make sense to keep the loops here because we'll filter args r_u and not export them automatically